src/workspace-data/data-table/entity-service/AttributeInput.ts

```diff
@@ -0,0 +1,307 @@
+import { TooltipTrigger } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import { Fragment, useEffect, useRef, useState } from 'react';
+import { div, fieldset, h, label, legend, span } from 'react-hyperscript-helpers';
+import { IdContainer, LabeledCheckbox, Link, RadioButton, Select } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import * as Utils from 'src/libs/utils';
+
+import { renderInputForAttributeType } from '../shared/AttributeInput';
+import { convertAttributeValue, getAttributeType } from './attribute-utils';
+
+interface AttributeTypeInputProps {
+  label?: string;
+  value: TypeOption;
+  onChange: (newValue: TypeOption) => void;
+  entityTypes?: string[];
+  defaultReferenceEntityType?: string;
+  showJsonTypeOption?: boolean;
+}
+
+interface BaseTypeOption {
+  type: string;
```

Comment:
i think we can restrict type field to union type for the known possible string values, 'string' | 'reference' | 'number', etc...  This would add more type safety and may ease the logic bits down below.